### PR TITLE
feat: upgrade pulsar to 0.9 API

### PR DESCRIPTION
**Pull request by yipeng1030**
what's changed

1. add topologies
 Pulsar supports three different cluster topologies, with two defined in clusterDefinition and one defined manually as creating the cluster:
- pulsar-basic-cluster
Configured in the pulsar-cluster/values.yaml file
This cluster topology includes three components: broker, zookeeper and bookkeeper
- pulsar-enhanced-cluster
Configured in the pulsar-cluster/values.yaml file
This cluster topology includes five components: broker, zookeeper, bookkeeper, proxy and bkrecovery
- Unspecified topology (i.e. topology: nil)
In this case, you can individually configure the proxy.enabled and bkrecovery.enabled options
This cluster approach allows more flexible composition of the required components
With all three topologies, you can configure serviceRef of external zookeeper service.
2. replace clusterDefinition and clusterVersion with componentDefinition and componentVerison.
3. update Vars definition like serviceRef and service.

Test
With all kinds of topologies, the cluster can run correctly.

### CODE REVIEW

**Comment by yipeng1030** [comment]
fix #686 as well.

File: addons/pulsar/templates/componentversion-bkrecovery.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        - pulsar-bkrecovery
+      releases:
+        - 2.11.2
+        - 3.0.2
+  releases:
+    - name: 2.11.2
+      changes:
+      serviceVersion: 2.11.2
+      images:
+        bookies-recovery: infracreate-registry.cn-zhangjiakou.cr.aliyuncs.com/apecloud/pulsar:2.11.2
```
Comment (nayutah) [inline addons/pulsar/templates/componentversion-bkrecovery.yaml:17]: switch aliyun image registry infracreate to apecloud, follow #716

**Comment by yipeng1030** [comment]
@leon-inf @Y-Rookie May I ask you to review this pr for kb 0.9? After test, basic features are completed.

File: addons-cluster/pulsar-cluster/templates/tests/benchmark.yaml [hunk 2]
```diff
@@ -85,7 +85,7 @@ metadata:
 spec:
   containers:
     - name: omb-k2k
-      image: docker.io/apecloud/omb:0.1
+      image: apecloud-registry.cn-zhangjiakou.cr.aliyuncs.com/apecloud/omb:0.1
```
Comment (nayutah) [inline addons-cluster/pulsar-cluster/templates/tests/benchmark.yaml:88]: make it configurable

File: addons/pulsar/templates/componentversion-bkrecovery.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        - pulsar-bkrecovery
+      releases:
+        - 2.11.2
+        - 3.0.2
+  releases:
+    - name: 2.11.2
+      changes:
+      serviceVersion: 2.11.2
+      images:
+        bookies-recovery: apecloud-registry.cn-zhangjiakou.cr.aliyuncs.com/apecloud/pulsar:2.11.2
```
Comment (nayutah) [inline addons/pulsar/templates/componentversion-bkrecovery.yaml:17]: make registry configurable

File: addons/pulsar/templates/componentversion-bkrecovery.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    - name: 3.0.2
+      changes:
+      serviceVersion: 3.0.2
+      images:
```
Comment (nayutah) [inline addons/pulsar/templates/componentversion-bkrecovery.yaml:21]: ditto

File: addons/pulsar/templates/componentversion-bookkeeper.yaml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    - compDefs:
+        - pulsar-bookkeeper
+      releases:
+        - 2.11.2
+        - 3.0.2
+  releases:
+    - name: 2.11.2
+      changes:
+      serviceVersion: 2.11.2
+      images:
```
Comment (nayutah) [inline addons/pulsar/templates/componentversion-bookkeeper.yaml:16]: ditto

File: addons/pulsar/templates/componentversion-broker.yaml [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    - compDefs:
+        - pulsar-broker
+      releases:
+        - 2.11.2
+        - 3.0.2
+  releases:
+    - name: 2.11.2
+      changes:
+      serviceVersion: 2.11.2
+      images:
```
Comment (nayutah) [inline addons/pulsar/templates/componentversion-broker.yaml:16]: ditto

File: addons/pulsar/templates/componentversion-proxy.yaml [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+    - compDefs:
+        - pulsar-proxy
+      releases:
+        - 2.11.2
+        - 3.0.2
+  releases:
+    - name: 2.11.2
+      changes:
+      serviceVersion: 2.11.2
+      images:
```
Comment (nayutah) [inline addons/pulsar/templates/componentversion-proxy.yaml:16]: ditto

File: addons/pulsar/templates/componentversion-zookeeper.yaml [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+    - compDefs:
+        - pulsar-zookeeper
+      releases:
+        - 2.11.2
+        - 3.0.2
+  releases:
+    - name: 2.11.2
+      changes:
+      serviceVersion: 2.11.2
+      images:
```
Comment (nayutah) [inline addons/pulsar/templates/componentversion-zookeeper.yaml:16]: ditto

**Review by nayutah — APPROVED** [review]

**Pull request merged**